go/processor/floria/processor_test.go

```diff
@@ -367,3 +367,28 @@ func TestProcessor_CallParameters(t *testing.T) {
 
 	}
 }
+
+func TestProcessor_SetUpAccessList(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	context := tosca.NewMockTransactionContext(ctrl)
+
+	transaction := tosca.Transaction{
+		Recipient: &tosca.Address{1},
+		AccessList: []tosca.AccessTuple{
+			{
+				Address: tosca.Address{2},
+				Keys:    []tosca.Key{{1}, {2}},
+			},
+		},
+	}
+
+	context.EXPECT().AccessStorage(*transaction.Recipient, tosca.Key{})
+	for _, contract := range getPrecompiledAddresses(tosca.R09_Berlin) {
+		context.EXPECT().AccessStorage(contract, tosca.Key{})
+	}
+	context.EXPECT().AccessStorage(tosca.Address{2}, tosca.Key{1})
+	context.EXPECT().AccessStorage(tosca.Address{2}, tosca.Key{2})
+
+	setUpAccessList(transaction, context, tosca.R09_Berlin)
+	ctrl.Finish()
```

Comment:
This should not be needed. If you initialize the controller with `t` it will add automatically an end-of-test callback to finalize the controller.